cmd/meroxa/root/apps/deploy.go

```diff
@@ -189,13 +201,189 @@ func (d *Deploy) createApplication(ctx context.Context, pipelineUUID, gitSha str
 	return nil
 }
 
+// uploadSource creates first a Dockerfile to then, package the entire folder which will be later uploaded
+// this should ignore .git files and fixtures/.
+func (d *Deploy) uploadSource(ctx context.Context, appPath, url string) error {
+	// Before creating a .tar.zip, we make sure it contains a Dockerfile.
+	err := turbine.CreateDockerfile(appPath)
+	if err != nil {
+		return err
+	}
+
+	dFile := fmt.Sprintf("turbine-%s.tar.gz", d.appName)
+
+	var buf bytes.Buffer
+	d.logger.Infof(ctx, "Packaging application located at %q...", appPath)
+	err = turbineCLI.CreateTarAndZipFile(appPath, &buf)
+	if err != nil {
+		return err
+	}
+
+	fileToWrite, err := os.OpenFile(dFile, os.O_CREATE|os.O_RDWR, os.FileMode(0777)) //nolint:gomnd
+	defer func(fileToWrite *os.File) {
+		err = fileToWrite.Close()
+		if err != nil {
+			panic(err.Error())
+		}
+	}(fileToWrite)
+
+	if err != nil {
+		panic(err)
```

Comment:
why are we panicking here?  i can see how it makes sense inside the defer function, but i was surprised to see it here and on 234

Reply:
@janelletavares you're right. I should deal with it different. I'll change it